std/net/curl.d

```diff
@@ -627,6 +627,61 @@ unittest
     assert(res == cast(ubyte[])[17, 27, 35, 41]);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *  perhaps this should live in std.uri
+ */
+private string enc()(in string[string] values)
+{
+    string enc;
+    foreach (k, v; values)
+    {
+        if (enc.length)
+            enc ~= '&';
+        import std.uri;
+        enc ~= encodeComponent(k) ~ "=" ~ encodeComponent(v);
+    }
+    return enc;
+}
+
+/** HTTP post content.
+ *
+ * Params:
+ * url = resource to post to
+ * postData = data to send as the body of the request. An associative array
+ *            of string is accepted and will be encoded using www-form-urlencoding
+ * conn = HTTP connection to use
+ *
+ * The template parameter $(D T) specifies the type to return. Possible values
+ * are $(D char) and $(D ubyte) to return $(D char[]) or $(D ubyte[]). If asking
+ * for $(D char), content will be converted from the connection character set
+ * (specified in HTTP response headers or FTP connection properties, both ISO-8859-1
+ * by default) to UTF-8.
+ *
+ * Example:
+ * ----
+ * import std.net.curl;
+ * auto content = post("d-lang.appspot.com/testUrl2", ["name1":"value1","name2":"value2"]);
+ * ----
+ *
+ * Returns:
+ * A T[] range containing the content of the resource pointed to by the URL.
+ *
+ * See_Also: $(LREF HTTP.Method)
+ */
+T[] post(T = char)(const(char)[] url, string[string] postData, HTTP conn = HTTP())
+{
+    return post(url,enc(postData),conn);
```

Comment:
Without config dfmt defaults options are phobos compatible. But if you use dfmt over the whole file...well you'll see, it might be better to correct this by hand.